src/sentry/integrations/github/tasks/link_all_repos.py

```diff
@@ -93,26 +77,31 @@ def link_all_repos(
                 lifecycle.record_halt(str(LinkAllReposHaltReason.RATE_LIMITED))
                 return
 
-            metrics.incr(f"{integration_key}.link_all_repos.api_error")
             raise
 
         integration_repo_provider = get_integration_repository_provider(integration)
 
-        # If we successfully create any repositories, we'll set this to True
-        success = False
-
+        repo_configs: list[dict[str, Any]] = []
+        missing_repos = []
         for repo in repositories:
             try:
-                config = get_repo_config(repo, integration_id)
-                integration_repo_provider.create_repository(
-                    repo_config=config, organization=rpc_org
-                )
-                success = True
+                repo_configs.append(get_repo_config(repo, integration_id))
             except KeyError:
-                continue
-            except RepoExistsError:
-                metrics.incr("sentry.integration_repo_provider.repo_exists")
+                missing_repos.append(repo)
                 continue
 
-        if not success:
-            lifecycle.record_halt(str(LinkAllReposHaltReason.REPOSITORY_NOT_CREATED))
+        try:
+            integration_repo_provider.create_repositories(
+                configs=repo_configs, organization=rpc_org
+            )
+        except RepoExistsError as e:
+            lifecycle.record_halt(
+                str(LinkAllReposHaltReason.REPOSITORY_NOT_CREATED),
+                {"missing_repos": e.repos, "integration_id": integration_id},
+            )
+
+        if missing_repos:
+            lifecycle.record_halt(
+                str(LinkAllReposHaltReason.REPOSITORY_NOT_CREATED),
+                {"missing_repos": missing_repos, "integration_id": integration_id},
+            )
```

Comment:
Are there any other partial failures cases we need to worry about?

What does `integration_repo_provider.create_repositories` do if any of these repo creations fail?

Reply:
The only case is where some fail and some succeed. Previously we would actually mark it as success if any repo was created, but here we only mark it as success if ALL repos are created.

If repo creation fails, we now raise an exception with the failed repos and mark it as a halt